src/Simplify.elm

```diff
@@ -1882,20 +1879,25 @@ multiplyChecks { parentRange, leftRange, rightRange, left, right } =
                                 [ Fix.removeRange (getRange ()) ]
                             ]
 
-                    else if value == 0 then
+                    else if number == 0 then
                         Just
-                            [ Rule.errorWithFix
-                                { message = "Multiplying by 0 equals 0"
-                                , details = [ "You can replace this value by 0." ]
+                            [ Rule.error
+                                { message = "Multiplication by 0 should be replaced"
+                                , details =
+                                    [ "Multiplying by 0 will turn finite numbers into 0 and keep NaN and (-)Infinity"
+                                    , "Most likely, multiplying by 0 was unintentional and you had a different factor in mind."
+                                    , """If you do want the described behavior, though, make your intention clear for the reader
+by explicitly checking for [`Basics.isNaN`](https://package.elm-lang.org/packages/elm/core/latest/Basics#isNaN)
+and [`Basics.isInfinite`](https://package.elm-lang.org/packages/elm/core/latest/Basics#isInfinite)"""
```

Comment:
There isn't any Markdown rendering in the details (that I can recall at least!), so I'm thinking it will render better like this:

```suggestion
                                    , """If you do want the described behavior, though, make your intention clear for the reader
by explicitly checking for `Basics.isNaN` and `Basics.isInfinite`."""
                                    , """Basics.isNaN: https://package.elm-lang.org/packages/elm/core/latest/Basics#isNaN
Basics.isInfinite: https://package.elm-lang.org/packages/elm/core/latest/Basics#isInfinite
```

(I'm open to suggestions on how to support Markdown/links/formatting in error details, but that's for another time :D )